openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -263,7 +263,25 @@ def get_libraries_for_user(user, org=None, library_type=None, text_search=None):
             Q(learning_package__description__icontains=text_search)
         )
 
-    return permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
+    filtered = permissions.perms[permissions.CAN_VIEW_THIS_CONTENT_LIBRARY].filter(user, qs)
+
+    if order:
+        order_query = 'learning_package__'
+        if '-' in order:
+            order_query = f"-{order_query}"
+            order = order.replace("-", "")
+
+        try:
+            filtered_ordered = filtered.order_by(f"{order_query}{order}")
+            # Evaluate the queryset to check if order field is a valid,
+            # otherwise if it is invalid and the queryset gets evaluated downstream
+            # the exception raised will not be handled here
+            if str(filtered_ordered.query):
+                return filtered_ordered
+        except FieldError as e:
+            log.exception(f"Error ordering libraries by {order}: {e}")
+
+    return filtered
```

Comment:
Not really a huge fan of evaluating the queryset just to check whether the provided `order` value is valid. However if I don't do it here and it gets evaluated later (like in `paginated_qs = self.paginate_queryset(queryset)` in the view) the error raised doesn't get handled. To avoid that I could add a `try/catch` there as well, just wondering if there is an even better approach?

Reply:
Thanks for the great suggestion! This approach is much more robust indeed, I needed to check the related_model `learning_package` but followed the same idea: [cfb9f55](https://github.com/openedx/edx-platform/pull/35005/commits/cfb9f558282095bce7f71e1cff13dad9432a6323)